libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventMapper.kt

```diff
@@ -0,0 +1,61 @@
+package com.mapbox.navigation.core.history.model
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.navigation.core.navigator.toLocation
+import com.mapbox.navigation.navigator.internal.ActiveGuidanceOptionsMapper.mapToActiveGuidanceMode
+import com.mapbox.navigation.navigator.internal.ActiveGuidanceOptionsMapper.mapToGeometriesCriteria
+import com.mapbox.navigator.HistoryRecord
+import com.mapbox.navigator.HistoryRecordType
+import com.mapbox.navigator.Waypoint
+
+internal class HistoryEventMapper {
+
+    fun map(historyRecord: HistoryRecord): HistoryEvent {
+        return when (historyRecord.type) {
+            HistoryRecordType.UPDATE_LOCATION -> mapUpdateLocation(historyRecord)
```

Comment:
```suggestion
            HistoryRecordType.UPDATE_LOCATION -> mapUpdateLocation(historyRecord.updateLocation!!)
```
so that we force cast in the same method that checks _if_ we can force cast. Same below.

Reply:
yeah i like it, thanks!